sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1376,24 +1377,19 @@ bool gnrc_netif_ipv6_wait_for_global_address(gnrc_netif_t *netif,
         } else {
             DEBUG_PUTS("gnrc_netif: got global address");
             has_global = true;
-            break;
         }
     }
 
-out:
     /* called with a given interface */
-    if (count == 0) {
-        msg_bus_t *bus = gnrc_netif_get_bus(netif, GNRC_NETIF_BUS_IPV6);
-        msg_bus_detach(bus, &subs[count]);
+    if (netif != NULL) {
+        _netif_bus_detach(netif, &subs[0]);
         return has_global;
     }
```

Comment:
```suggestion
    }
    else {
```

IMHO the less early returns, the better (and in case the optimizer gets confused it might also lead to smaller code size). Also I think that makes the code easier to understand: The if branch handles the "`netif` is set" case the else branch the "`netif` is unset".